app/src/main/java/net/gsantner/markor/format/markdown/MarkdownAutoFormat.java

```diff
@@ -13,17 +13,14 @@
 import android.text.Spanned;
 
 import java.util.regex.Matcher;
+import java.util.Arrays;
 
 public class MarkdownAutoFormat implements InputFilter {
     @Override
     public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
         try {
-            if (end - start == 1
-                    && start < source.length()
-                    && dstart <= dest.length()) {
-                char newChar = source.charAt(start);
-
-                if (newChar == '\n') {
+            if (start < source.length() && dstart <= dest.length()) {
+                if ((source.charAt(start) == '\n') || (source.charAt(end - 1) == '\n')) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5254231)